libdevcore/Worker.cpp

```diff
@@ -118,7 +118,8 @@ void Worker::terminate()
 	std::unique_lock<Mutex> l(x_work);
 	if (m_work)
 	{
-		m_state.exchange(WorkerState::Killing);
+		if (m_state.exchange(WorkerState::Killing) == WorkerState::Killing)
```

Comment:
Check also stopWorking() - it has similar pattern.

Reply:
I added a similar change there.